### PR TITLE
GH-250 Made remote helm repository arg helm_charts_base_url optional

### LINKED ISSUE #250: Remote helm repository arg helm_charts_base_url is mistakenly required
**Describe the bug**
`artifactory_remote_helm_repository` mistakenly requires `helm_charts_base_url` to be set, even though the doc says it's optional (and an empty value is allowed in the UI).  

**Requirements for an issue**
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue)

```
resource "artifactory_remote_helm_repository" "default_chartcenter" {
  key = "default-chartcenter"
  url = "https://repo.chartcenter.io/"
  # todo:  we don't really use this but it's a required arg that can't be empty (even though 
  # empty is fine in the UI and the docs say optional)
  helm_charts_base_url = "https://repo.chartcenter.io/"
}
```

- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
Cloud

- [x] Your version of terraform
`1.0.11` w/artifactory provider `2.6.23`

**Expected behavior**
Should behave as [documented](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/artifactory_remote_helm_repository).  The provider should allow the field to be empty (or not provided).  Doc for reference:

> # Argument Reference
> Arguments have a one to one mapping with the JFrog API. All generic repo arguments are supported, in addition to:
> 
> `key` - (Required) The repository identifier. Must be unique system-wide
> `helm_charts_base_url` - (Optional) - No documentation is available. Hopefully you know what this means

### PR DESCRIPTION
**Pull request by maheshjfrog**
Fixes #250

### CODE REVIEW

File: pkg/artifactory/resource_artifactory_remote_helm_repository.go [hunk 1]
```diff
@@ -7,9 +7,10 @@ import (
 
 var helmRemoteSchema = mergeSchema(baseRemoteSchema, map[string]*schema.Schema{
 	"helm_charts_base_url": {
-		Type:         schema.TypeString,
-		Required:     true,
-		ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+		Type:             schema.TypeString,
+		Optional:         true,
+		Default:          "",
+		ValidateDiagFunc: validation.ToDiagFunc(validation.Any(validation.StringIsEmpty, validation.IsURLWithHTTPorHTTPS)),
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_remote_helm_repository.go:13]: So, it can be empty, or it can be http?

**Review by chb0github — APPROVED** [review]

**Review by chb0github — APPROVED** [review]

File: pkg/artifactory/resource_artifactory_remote_helm_repository.go (hunk 1 above)
Comment (maheshjfrog) [reply pkg/artifactory/resource_artifactory_remote_helm_repository.go]: Yes. 
On negative cases, the first validation is shown. So, Made IsURLWithHTTPorHTTPS as the first validation check followed by StringIsEmpty check.

Tested following scenarios:
Blank input to helm_charts_base_url argument
Without helm_charts_base_url argument
Valid helm_charts_base_url argument (http and https)
Invalid helm_charts_base_url argument (like http1)

Comment (alexhung) [reply pkg/artifactory/resource_artifactory_remote_helm_repository.go]: Please add these test cases to `resource_artifactory_remote_repository_test.go` 😄

**Review by alexhung — APPROVED** [review]
Looks fine. Bonus points for adding test cases.

**Pull request merged**